src/helm/proxy/clients/http_client.py

```diff
@@ -0,0 +1,151 @@
+        cache_key = asdict(request)
+        # This needs to match whatever we define in pedantic
+        if request.embedding:
+            return EMBEDDING_UNAVAILABLE_REQUEST_RESULT
+
+        # Only a single stop sequence is supported as we can only pass in a single value for `eos_token_id`
+        if len(request.stop_sequences) > 1:
+            raise ValueError("More than one stop sequence is not supported.")
+
+        raw_request = {
+            "prompt": request.prompt,
+            "temperature": 1e-7 if request.temperature == 0 else request.temperature,
+            "num_return_sequences": request.num_completions,
+            "max_new_tokens": request.max_tokens,
+            "top_p": request.top_p,
+            "echo_prompt": request.echo_prompt,
+            "top_k_per_token": request.top_k_per_token,
+            "stop_sequences": request.stop_sequences,
+        }
+
+        try:
+
+            def do_it():
+                url = f"{self.base_url}:{self.port}/process"
+                response = requests.post(url, json=raw_request, timeout=self.timeout)
+                response.raise_for_status()
+                response_data = response.json()
+                return response_data
+
+            if self.do_cache:
+                response, cached = self.cache.get(cache_key, wrap_request_time(do_it))
+            else:
+                response, cached = do_it(), False
+
+            tokens = [
+                Token(
+                    text=token["text"], logprob=token["logprob"], top_logprobs=token["top_logprob"]
+                )
+                for token in response["tokens"]
+            ]
+            completions = [
+                Sequence(text=response["text"], logprob=response["logprob"], tokens=tokens)
+            ]
+
+            return RequestResult(
+                success=True,
+                cached=cached,
+                error=None,
+                completions=completions,
+                embedding=[],
+                request_time=response["request_time"],
+            )
+        except requests.exceptions.RequestException as e:
+            error: str = f"Request error: {e}"
+            return RequestResult(
+                success=False, cached=False, error=error, completions=[], embedding=[]
+            )
+
+    def tokenize(self, request: TokenizationRequest) -> TokenizationRequestResult:
+        cache_key = asdict(request)
+        raw_request = {
+            "text": request.text,
+            "truncation": request.truncation,
+            "max_length": request.max_length,
+        }
+
+        try:
+
+            def do_it():
+                url = f"http://localhost:{self.port}/tokenize"
+                response = requests.post(url, json=raw_request)
+                response.raise_for_status()
+                response_data = response.json()
+                return response_data
+
+            result, cached = self.cache.get(cache_key, wrap_request_time(do_it))
```

Comment:
If `do_cache` is `False`, we probably shouldn't cache the tokenization request either.